lib/connectors/memory.js

```diff
@@ -383,7 +399,9 @@ Memory.prototype._castPropertyValue = function(prop, val, props) {
       break;
     case 'ModelConstructor':
       for (var subProp in val) {
-        val[subProp] = this._castPropertyValue(subProp, val[subProp], propType.definition.properties);
```

Comment:
Is this checking for something different from the above checks? Is this suppressing cases that would normally trigger the checks in `_castPropertyValue`?